api/src/main/java/com/github/retrooper/packetevents/util/VectorMutable3d.java

```diff
@@ -0,0 +1,318 @@
+     */
+    public double x;
+    /**
+     * Y (coordinate/angle/whatever you wish)
+     */
+    public double y;
+    /**
+     * Z (coordinate/angle/whatever you wish)
+     */
+    public double z;
+
+    /**
+     * Default constructor setting all coordinates/angles/values to their default values (=0).
+     */
+    public VectorMutable3d() {
+        this.x = 0.0;
+        this.y = 0.0;
+        this.z = 0.0;
+    }
+
+    /**
+     * Constructor allowing you to set the values.
+     *
+     * @param x X
+     * @param y Y
+     * @param z Z
+     */
+    public VectorMutable3d(double x, double y, double z) {
+        this.x = x;
+        this.y = y;
+        this.z = z;
+    }
+
+    /**
+     * Constructor allowing you to specify an array.
+     * X will be set to the first index of an array(if it exists, otherwise 0).
+     * Y will be set to the second index of an array(if it exists, otherwise 0).
+     * Z will be set to the third index of an array(if it exists, otherwise 0).
+     *
+     * @param array Array.
+     */
+    public VectorMutable3d(double[] array) {
+        if (array.length > 0) {
+            x = array[0];
+        } else {
+            x = 0;
+            y = 0;
+            z = 0;
+            return;
+        }
+
+        if (array.length > 1) {
+            y = array[1];
+        } else {
+            y = 0;
+            z = 0;
+            return;
+        }
+
+        if (array.length > 2) {
+            z = array[2];
+        } else {
+            z = 0;
+        }
+    }
+
+    public static VectorMutable3d read(PacketWrapper<?> wrapper) {
+        double x = wrapper.readDouble();
+        double y = wrapper.readDouble();
+        double z = wrapper.readDouble();
+        return new VectorMutable3d(x, y, z);
+    }
+
+    public static void write(PacketWrapper<?> wrapper, VectorMutable3d vector) {
+        wrapper.writeDouble(vector.x);
+        wrapper.writeDouble(vector.y);
+        wrapper.writeDouble(vector.z);
+    }
+
+    public static VectorMutable3d decode(NBT tag, ClientVersion version) {
+        NBTList<?> list = (NBTList<?>) tag;
+        double x = ((NBTNumber) list.getTag(0)).getAsDouble();
+        double y = ((NBTNumber) list.getTag(1)).getAsDouble();
+        double z = ((NBTNumber) list.getTag(2)).getAsDouble();
+        return new VectorMutable3d(x, y, z);
+    }
+
+    public static NBT encode(VectorMutable3d vector3d, ClientVersion version) {
+        NBTList<NBTDouble> list = new NBTList<>(NBTType.DOUBLE, 3);
+        list.addTag(new NBTDouble(vector3d.x));
+        list.addTag(new NBTDouble(vector3d.y));
+        list.addTag(new NBTDouble(vector3d.z));
+        return list;
+    }
+
+    @Override
+    public double getX() {
+        return x;
+    }
+
+    @Override
+    public double getY() {
+        return y;
+    }
+
+    @Override
+    public double getZ() {
+        return z;
+    }
+
+    public boolean equals(VectorMutable3d other) {
```

Comment:
Shouldnt the parameter be the interface? (Same for every other method in this class)